tests/query/snql/test_query.py

```diff
@@ -2,14 +2,15 @@
 
 from snuba import state
 from snuba.datasets.factory import get_dataset
+from snuba.query.conditions import binary_condition
 from snuba.query.expressions import Column, FunctionCall, Literal
 from snuba.query.logical import OrderBy, OrderByDirection, Query, SelectedExpression
 from snuba.query.snql import parse_snql_query
 
 
 test_cases = [
```

Comment:
I think you should also start adding some negative tests, i.e. queries that shouldn't work.

For example, I'm curious what happens if I omit spaces from my query. Based on the regex, spaces seem to be totally optional.

Is `MATCH(blah)WHEREa<3COLLECT4-5` a valid query?

Reply:
Hi, I've started adding some test cases that serve as invalid queries that hopefully begin to illustrate what sorts of queries cannot be parsed and why. Would appreciate any further suggestions to add more cases!